cai_causal_graph/time_series_causal_graph.py

```diff
@@ -769,6 +776,9 @@ def delete_node(self, identifier: NodeLike):
 
         See `cai_causal_graph.causal_graph.CausalGraph.delete_node` for more details.
         """
+        node = self.get_node(self._NodeCls.identifier_from(identifier))
+        assert isinstance(node, TimeSeriesNode)  # for linting
+        self.remove_node_from_cache(node)
```

Comment:
Move this logic into `delete_node` because currently if I do `delete_node` then the node is still in the cache. Add a test for that too pls.

Reply:
So this is currently in `delete_node`, `remove_node` in `causal_graph` calls this so should work either way? (have added test for this)

Reply:
Ah sorry - im being dumb. All good - was looking at the wrong thing.